### PR TITLE
Incorporate shorter durations for other outcomes

### PR DESCRIPTION
**Pull request by collijk**
### Description

*Category*: feature
*JIRA issue*: [MIC-2970](https://jira.ihme.washington.edu/browse/MIC-2970)
[Research reference](https://github.com/ihmeuw/vivarium_research/pull/827)

Adds duration sampling conditional on pregnancy outcome.

NOTE: I think this will need an update to the birth recorder and
potentially to other observers.

### Verification and Testing
TODO

### CODE REVIEW

**Review by albrja — APPROVED** [review]

File: src/vivarium_gates_iv_iron/components/pregnancy.py [hunk 1]
```diff
@@ -225,20 +231,34 @@ def _sample_new_pregnant(self, index: pd.Index, event_time: pd.Timestamp) -> pd.
...
+
+        child_cols = self.new_children.columns_created
+        newly_pregnant.loc[live_or_still_birth, child_cols] = (
+            self.new_children(live_or_still_birth)
+        )
+
+        low, high = DURATIONS.DETECTION, DURATIONS.PARTIAL_TERM
+        draw = self.randomness.get_draw(newly_pregnant.index, additional_key='pregnancy_duration')
+        pregnancy_duration = pd.to_timedelta(
+            7 * low + (high - low) * draw, unit='days'
```
Comment (alibow) [inline src/vivarium_gates_iv_iron/components/pregnancy.py:255]: Should this read `7 * (low + (high - low) * draw)`? In other words, is it missing the parenthetical to scale the whole thing to days?

File: src/vivarium_gates_iv_iron/components/pregnancy.py [hunk 2]
```diff
@@ -225,20 +231,34 @@ def _sample_new_pregnant(self, index: pd.Index, event_time: pd.Timestamp) -> pd.
...
+        other_outcome = pregnancy_outcome[pregnancy_outcome == models.OTHER_OUTCOME].index
+        live_or_still_birth = pregnancy_outcome.index.difference(other_outcome)
+
+        child_cols = self.new_children.columns_created
+        newly_pregnant.loc[live_or_still_birth, child_cols] = (
+            self.new_children(live_or_still_birth)
+        )
+
+        low, high = DURATIONS.DETECTION, DURATIONS.PARTIAL_TERM
+        draw = self.randomness.get_draw(newly_pregnant.index, additional_key='pregnancy_duration')
```
Comment (alibow) [inline src/vivarium_gates_iv_iron/components/pregnancy.py:253]: Does this imply pregnancy duration for the other outcome is determined a the draw level? I had envisioned it varying at the simulant level.

Comment (alibow) [reply src/vivarium_gates_iv_iron/components/pregnancy.py:253]: Sorry that wasn't more clear!

Comment (collijk) [reply src/vivarium_gates_iv_iron/components/pregnancy.py:253]: No, this is done at the simulant level.  This just gets a uniform random number for every person in the index.

**Pull request closed without merging**